prov/efa/src/efa_hmem.c

```diff
@@ -409,42 +372,18 @@ int efa_domain_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem
  */
 int efa_domain_hmem_info_init_all(struct efa_domain *efa_domain)
 {
-	int ret, err;
+	int ret = 0;
+	enum fi_hmem_iface ifaces[4] = {FI_HMEM_SYSTEM, FI_HMEM_CUDA,
+	                                FI_HMEM_NEURON, FI_HMEM_SYNAPSEAI};
 
 	if(g_device_cnt <= 0) {
 		return -FI_ENODEV;
 	}
 
 	memset(efa_domain->hmem_info, 0, OFI_HMEM_MAX * sizeof(struct efa_hmem_info));
 
-	ret = 0;
-
-	err = efa_domain_hmem_info_init_system(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the System hmem_info struct! err: %d\n",
-			 err);
-	}
-
-	err = efa_domain_hmem_info_init_cuda(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Cuda hmem_info struct! err: %d\n",
-			 err);
-	}
-
-	err = efa_domain_hmem_info_init_neuron(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Neuron hmem_info struct! err: %d\n",
-			 err);
-	}
-
-	err = efa_domain_hmem_info_init_synapseai(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Synapseai hmem_info struct! err: %d\n",
-			 err);
+	for (int i = 0; i < sizeof(ifaces) / sizeof(enum fi_hmem_iface); ++i) {
+		efa_domain_hmem_info_init_iface(efa_domain, ifaces[i]);
 	}
```

Comment:
There's also a convenience macro for this in `efa_hmem.h`

https://github.com/ofiwg/libfabric/blob/5b08723bf29a914d045554c0942e6d058a58397e/prov/efa/src/efa_hmem.h#L10-L11